File: frontend/types/isp_station.ts
export interface ISPStation {
  id: string;
  name: string;
  description?: string;
  organization: {
    id: string;
    name: string;
  };
  location: string;
  buildingType: string;
  notes?: string;
  status: 'ACTIVE' | 'INACTIVE' | 'MAINTENANCE';
  coordinates?: string;
  createdAt: string;
  updatedAt: string;
}

export interface ISPStationsResponse {
  stations: {
    success: boolean;
    message: string;
    stations: ISPStation[];
    totalCount: number;
  };
}

export interface StationFilterOptions {
  page: number;
  pageSize: number;
  sortBy: string;
  sortDirection: 'asc' | 'desc';
  search: string;
  filterStatus?: string;
}

File: backend/app/models/isp_station.py
from datetime import datetime
from typing import Optional
from app.schemas.isp_station import StationStatus, BuildingType
import strawberry

@strawberry.type
class DBISPStation:
    _id: str
    name: str
    description: Optional[str]
    organizationId: str
    location: str
    buildingType: BuildingType
    notes: Optional[str]
    status: StationStatus
    coordinates: Optional[str]
    createdAt: datetime
    updatedAt: datetime